pkg/odo/cli/component/ui/ui.go

```diff
@@ -0,0 +1,212 @@
+	for _, option := range options {
+		if option.Name == selectedComponentType {
+			sort.Strings(option.NonHiddenTags)
+			return option.NonHiddenTags
+		}
+	}
+	glog.V(4).Infof("Selected component type %s was not part of the catalog images", selectedComponentType)
+	return []string{}
+}
+
+// SelectSourceType lets the user select a specific occlient.CreateType in a prompty
+func SelectSourceType(sourceTypes []occlient.CreateType) occlient.CreateType {
+	options := make([]string, len(sourceTypes))
+	for i, sourceType := range sourceTypes {
+		options[i] = fmt.Sprint(sourceType)
+	}
+
+	var selectedSourceType string
+	prompt := &survey.Select{
+		Message: "Which input type would wish to use for the component",
+		Options: options,
+	}
+	err := survey.AskOne(prompt, &selectedSourceType, survey.Required)
+	ui.HandleError(err)
+
+	for _, sourceType := range sourceTypes {
+		if selectedSourceType == fmt.Sprint(sourceType) {
+			return sourceType
+		}
+	}
+	glog.V(4).Infof("Selected source type %s was not part of the source type options", selectedSourceType)
+	return occlient.NONE
+}
+
+// EnterInputTypePath allows the user to specify the path on the filesystem in a prompt
+func EnterInputTypePath(inputType string, defaultPath string) string {
+	var path string
+	prompt := &survey.Input{
+		Message: fmt.Sprintf("Where does the %s input for the component reside on the local file system", inputType),
+	}
+	if len(defaultPath) > 0 {
+		prompt.Default = defaultPath
+	}
+	err := survey.AskOne(prompt, &path, survey.Required)
+	ui.HandleError(err)
+	return path
+}
+
+// EnterComponentName allows the user to specify the component name in a prompt
+func EnterComponentName(defaultName string) string {
+	var path string
+	prompt := &survey.Input{
+		Message: "How would you wish to name the new component",
+		Default: defaultName,
+	}
+	err := survey.AskOne(prompt, &path, survey.Required)
+	ui.HandleError(err)
+	return path
+}
+
+// EnterGitInfo will display two prompts, one of the URL of the project and one of the ref
+func EnterGitInfo() (string, string) {
+	gitURL := enterGitInputTypePath()
+	gitRef := enterGitRef("master")
+
+	return gitURL, gitRef
+}
+
+func enterGitInputTypePath() string {
+	var path string
+	prompt := &survey.Input{
+		Message: "What is the URL of the git repository you would wish the new component to use",
+	}
+	err := survey.AskOne(prompt, &path, survey.Required)
+	ui.HandleError(err)
+	return path
+}
+
+func enterGitRef(defaultRef string) string {
+	var path string
+	prompt := &survey.Input{
+		Message: "What git ref (branch, tag, commit) would you wish to use",
+		Default: defaultRef,
+	}
+	err := survey.AskOne(prompt, &path, survey.Required)
+	ui.HandleError(err)
+	return path
+}
+
+// SelectYesNo displays a given message and asks the user to respond with yes or no and returns true of the answer is Yes
+func SelectYesNo(message string) bool {
```

Comment:
There's a component that does that: `survey.Confirm`. See https://github.com/redhat-developer/odo/blob/master/pkg/odo/cli/service/ui/ui.go#L262-L265

Reply:
I'll check it out, thanks!

Reply:
Used it, thanks